src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -69,7 +70,7 @@ public abstract class StringConfigurationSource : IConfigurationSource
         {
             string value = GetString(key);
 
-            return double.TryParse(value, out double result)
+            return double.TryParse(value, NumberStyles.Any, CultureInfo.InvariantCulture, out double result)
```

Comment:
Probably less of an issue with integers, but should we do the same with `GetInt32()` above?

Reply:
Currently non of the int.TryParse is using that overload, but all of the double ones were, except the fixed one. It might introduce new code pattern.
Only difference what is see is NumberGroupSeparator  (In case of InvariantCulture - ',') but it's usage is disabled with NumberStyles.Integer (Integer  = AllowLeadingWhite | AllowTrailingWhite | AllowLeadingSign) which is used by non overloaded method. 
So I think this change might cause more confusion and introduce new risks. (For eg specifying NumberStyles.Any which could lead to incorrect parsing).

Reply:
I agree with @RassK . The only other place I am aware of which could have a similar issue is parsing `DateTime`. All usages of `DataTime.Parse` and `Convert.ToDateTime` are looking goood